crates/pathfinder/src/ethereum/contract.rs

```diff
@@ -1,14 +1,53 @@
-use std::str::FromStr;
+// use std::str::FromStr;
 
 use web3::ethabi::{Contract, Event, Function};
 use web3::types::H160;
 
-/// The address of Starknet's core contract (proxy).
-const CORE_PROXY_ADDR: &str = "0xde29d060D45901Fb19ED6C6e959EB22d8626708e";
-/// The address of Starknet's general purpose solver contract.
-const GPS_ADDR: &str = "0x5EF3C980Bf970FcE5BbC217835743ea9f0388f4F";
-/// The address of Starknet's memory page contract.
-const MEMPAGE_ADDR: &str = "0x743789ff2fF82Bfb907009C9911a7dA636D34FA7";
+use crate::ethereum::Chain;
+
+pub struct StarknetCoreAddress(pub H160);
+pub struct StarknetGpsAddress(pub H160);
+pub struct StarknetMempageAddress(pub H160);
```

Comment:
I'm not sure these are really necessary, since they're already namespaced by the struct. Think I'll remove them.